src/twisted/scripts/trial.py

```diff
@@ -477,18 +510,22 @@ class Options(_BasicOptions, usage.Options, app.ReactorSelectionMixin):
 
     def opt_jobs(self, number):
         """
-        Number of local workers to run, a strictly positive integer.
+        Number of local workers to run, a strictly positive integer or 'auto'
+        to spawn one worker for each available CPU.
         """
-        try:
-            number = int(number)
-        except ValueError:
-            raise usage.UsageError(
-                "Expecting integer argument to jobs, got '%s'" % number
-            )
-        if number <= 0:
-            raise usage.UsageError(
-                "Argument to jobs must be a strictly positive integer"
-            )
+        if number == "auto":
```

Comment:
this looks good

one option to write this with a smalle diff and less indentation

```
if number == "auto":
    self["jobs"] = _autoJobs()
    return
```